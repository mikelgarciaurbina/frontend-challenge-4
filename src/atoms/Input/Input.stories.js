import React from 'react';
import { action } from '@storybook/addon-actions';

import Input from './index';

export default {
  title: 'Input',
  component: Input,
};

export const Default = () => (
  <Input
    onChange={action('input change')}
    onKeyUp={action('input keyUp')}
    placeholder="Type here..."
  />
);

export const WithDefaultValue = () => (
  <Input
    onChange={action('input change')}
    onKeyUp={action('input keyUp')}
    placeholder="Type here..."
    value="test"
  />
);
