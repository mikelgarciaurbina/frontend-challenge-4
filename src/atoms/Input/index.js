import React from 'react';
import PropTypes from 'prop-types';

export default function Input({ onChange, onKeyUp, placeholder, value }) {
  return (
    <input
      className="landbot-input"
      onChange={onChange}
      onKeyUp={onKeyUp}
      placeholder={placeholder}
      type="text"
      value={value}
    />
  );
}
Input.propTypes = {
  onChange: PropTypes.func,
  onKeyUp: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.string,
};
Input.defaultProps = {
  onChange() {},
  onKeyUp() {},
  placeholder: '',
  value: '',
};
