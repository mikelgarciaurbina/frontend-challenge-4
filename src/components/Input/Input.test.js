import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Input from './index';

configure({ adapter: new Adapter() });

describe('<Input />', () => {
  it('should render 1 input', () => {
    const wrapper = shallow(<Input />);

    expect(wrapper.find('input').length).toBe(1);
  });

  it('should be a text type input', () => {
    const wrapper = shallow(<Input />);

    expect(wrapper.find('input').props().type).toBe('text');
  });

  it('should render with landbot-input class name', () => {
    const wrapper = shallow(<Input />);

    expect(wrapper.find('input').props().className).toBe('landbot-input');
  });

  it('should render the input placeholder', () => {
    const wrapper = shallow(<Input placeholder="test" />);

    expect(wrapper.find('input').props().placeholder).toBe('test');
  });

  it('should render the input value', () => {
    const wrapper = shallow(<Input value="test" />);

    expect(wrapper.find('input').props().value).toBe('test');
  });

  it('should call to onChange on type in Input', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(<Input onChange={mockCallback} />);

    wrapper.find('input').simulate('change', { target: { value: 'test' } });

    expect(mockCallback).toBeCalledWith({ target: { value: 'test' } });
  });

  it('should call to onKeyUp on key up in Input', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(<Input onKeyUp={mockCallback} />);

    wrapper.find('input').simulate('keyUp', { key: 'Enter' });

    expect(mockCallback).toBeCalledWith({ key: 'Enter' });
  });

  it('Matches snapshot', () => {
    const wrapper = shallow(
      <Input
        placeholder="test"
        value="test"
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
