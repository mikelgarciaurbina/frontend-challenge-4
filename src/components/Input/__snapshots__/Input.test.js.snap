// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Matches snapshot 1`] = `
<input
  className="landbot-input"
  onChange={[Function]}
  onKeyUp={[Function]}
  placeholder="test"
  type="text"
  value="test"
/>
`;
